DuckDuckGo/LottieView.swift

```diff
@@ -22,35 +22,71 @@ import Lottie
  
 struct LottieView: UIViewRepresentable {
     
+    struct LoopWithIntroTiming {
+        let skipIntro: Bool
+        let introStartFrame: AnimationFrameTime
+        let introEndFrame: AnimationFrameTime
+        let loopStartFrame: AnimationFrameTime
+        let loopEndFrame: AnimationFrameTime
+    }
+
+    enum LoopMode {
+        case mode(LottieLoopMode)
+        case withIntro(LoopWithIntroTiming)
+    }
+
     let lottieFile: String
     let delay: TimeInterval
     var isAnimating: Binding<Bool>
-        
+    private let loopMode: LoopMode
+
     let animationView = AnimationView()
     
-    init(lottieFile: String, delay: TimeInterval = 0, isAnimating: Binding<Bool> = .constant(true)) {
+    init(lottieFile: String, delay: TimeInterval = 0, loopMode: LoopMode = .mode(.playOnce), isAnimating: Binding<Bool> = .constant(true)) {
         self.lottieFile = lottieFile
         self.delay = delay
         self.isAnimating = isAnimating
+        self.loopMode = loopMode
     }
 
     func makeUIView(context: Context) -> some AnimationView {
         animationView.animation = Animation.named(lottieFile)
         animationView.contentMode = .scaleAspectFit
         animationView.clipsToBounds = false
-        
+
+        switch loopMode {
+        case .mode(let lottieLoopMode): animationView.loopMode = lottieLoopMode
+        case .withIntro: break
+        }
+
         return animationView
     }
  
     func updateUIView(_ uiView: UIViewType, context: Context) {
+        if uiView.isAnimationPlaying, !isAnimating.wrappedValue {
+            uiView.stop()
+            return
+        }
```

Comment:
If the animation is playing but the wrapped value is false, then stop the view from animating.